buf/registry/module/v1beta1/tag_service.proto

```diff
@@ -39,6 +39,12 @@ service TagService {
   rpc CreateTags(CreateTagsRequest) returns (CreateTagsResponse) {
     option idempotency_level = IDEMPOTENT;
   }
+  // Create new Tags on a Module, and Update existing Tags on a Module.
+  //
+  // This operation is atomic. Either all Tags are created/updated or an error is returned.
+  rpc CreateOrUpdateTags(CreateOrUpdateTagsRequest) returns (CreateOrUpdateTagsResponse) {
```

Comment:
My question from [#33](https://github.com/bufbuild/registry-proto/pull/33#discussion_r1393337177) remains: is there any real downside to the non-atomic `Create` and then `Update` approach?

Reply:
The downside here is that `Update` should only be used when the Tag already exist, so from the client perspective you'd need to `Get` the tag first to know if you should invoke `Create` or `Update`.